src/components/pages/roster-page/roster-page.tsx

```diff
@@ -237,9 +238,12 @@ export const RosterPage = () => {
       return;
     }
 
-    dispatch(Roster.upload(e.target.files[0], async (count, message) => {
-      if (count < 0) {
-        const msg = `An error occurred while uploading the roster: ${message || 'Please verify the roster data.'}`;
+    dispatch(Roster.upload(e.target.files[0], async (response, message) => {
+      if (response.errors?.length || message) {
+        const msg = (response.errors ?? []).reduce((result, error) => {
+          return `${result}<br/>${error.error}${error.column ? ` on '${error.column.displayName}'` : ''} (${error.edipi})`;
+        }, `An error occurred while uploading the roster: ${message || 'Please verify the roster data.'}`)
```

Comment:
I think there's too much going on in these three lines. Using reduce for string building doesn't seem great to me (reduce almost always makes things more confusing, since you have to look at the end of the function then look back at the top in order to follow the logic). I'd much rather have things flow from top to bottom like the end text will. I think this sort of approach, while using more lines, is more natural to read...
```suggestion
        let msg = `An error occurred while uploading the roster: ${message || 'Please verify the roster data.'}`;
        if (response.errors) {
          for (const error of response.errors) {
            const errorDetails = error.column ? ` on '${error.column.displayName}'` : '';
            msg += `<br/>${error.error}${errorDetails} (${error.edipi})`;
          }
        }
```